pkg/image/admission/imagepolicy/rules/rules.go

```diff
@@ -91,37 +94,48 @@ func matchImageCondition(condition *api.ImageCondition, integrated RegistryMatch
 // Use matchImageCondition to apply invertMatch rules.
 func matchImageConditionValues(rule *api.ImageCondition, integrated RegistryMatcher, attrs *ImagePolicyAttributes) bool {
 	if rule.MatchIntegratedRegistry && !(attrs.IntegratedRegistry || integrated.Matches(attrs.Name.Registry)) {
+		glog.V(5).Infof("image registry %v does not match integrated registry", attrs.Name.Registry)
 		return false
 	}
 	if len(rule.MatchRegistries) > 0 && !hasAnyMatch(attrs.Name.Registry, rule.MatchRegistries) {
+		glog.V(5).Infof("image registry %v does not match registries from rule: %#v", attrs.Name.Registry, rule.MatchRegistries)
 		return false
 	}
 
 	// all subsequent calls require the image
 	image := attrs.Image
 	if image == nil {
 		if rule.SkipOnResolutionFailure {
-			return true
+			glog.V(5).Infof("rule does not match because image did not resolve and SkipOnResolutionFailure is true")
+			// Likely we will never get here (see: https://github.com/openshift/origin/blob/4f709b48f8e52e8c6012bd8b91945f022a437a6a/pkg/image/admission/imagepolicy/rules/accept.go#L99-L103)
+			// but if we do, treat the condition as not matching since we are supposed to skip this rule on resolution failure.
+			return false
```

Comment:
this is the controversial change.  I do not see how we even reach this point in the code(because as I note in the comment, the code that calls matchConditions has already filtered out cases of image==nil && skipOnResolutionFailure), but if someone else started using this function, I think this should be returning false.

The godoc for SkipOnResolutionFailure says:
```
	// SkipOnResolutionFailure allows the subsequent conditions to be bypassed if the integrated registry does
	// not have access to image metadata (no image exists matching the image digest).
```

So if the condition is supposed to be bypassed, then matchCondition should be returning "false".

Reply:
Arguably this check should even occur *before* we check if the resolved image matches the integrated registries and/or the rule registries.  But again it's moot given the other code that results in bypassing all of this.